libraries/SuplaDevice/SuplaDevice.cpp

```diff
@@ -171,7 +174,14 @@ bool SuplaDeviceClass::begin(char GUID[SUPLA_GUID_SIZE],
 
   setString(Supla::Channel::reg_dev.Email, email, SUPLA_EMAIL_MAXSIZE);
   setString(Supla::Channel::reg_dev.ServerName, Server, SUPLA_SERVER_NAME_MAXSIZE);
-
+  
+  /* protection from default parameters value mismatch */ 
+  /* if someone has used version parameter directly */
+  if (port == version)
+	this->port = 2015;
+  else 
+	this->port = port;
+  
```

Comment:
I think it will be better to move port and server address setting to Network class.
Some methods like:
setServerAddres( ...) 
setServerPort(...)

Then we can use ssl connection on ESP by default (so port 2016 will be default for ESP and for other network interfaces we'll have 2015 with insecure connection).

Reply:
At the moment, the Network class from the user's perspective is only responsible for connecting to the WiFi network. The supla server address is forwarded to SuplaDevice and if the address is there than the port is too. If we want to change it, we have to discard SuplaDevice (address and port at the moment) and transfer those to the Network class initialization.
As it is now it is more logical: we pass Supla parameters (including server address and port) in SuplaDevice class, and those regarding wifi in the class inheriting from Network. Your call but it will change the logic.

Reply:
You are right. However I already extracted a lot from SuplaDevice class and soon it will be a tiny object ;).
Then maybe other approach:
1. Remove port from begin() method
2. Pass port to network interface classes - if port is -1 then use defaults (2015 for arduino board classes and 2016 with ssl for ESP based classes) and make it dependent on "setInsecure"/"setSecure" methods